src/amber/router/pipe/pipeline.cr

```diff
@@ -23,7 +23,20 @@ module Amber
         if context.websocket?
           context.process_websocket_request
         else
-          @drain[context.valve].call(context) if @drain[context.valve]
+          begin
+            @drain[context.valve].call(context) if @drain[context.valve]
+          rescue e
```

Comment:
We can do a catch all here and initialize an exception from the Amber exceptions. We can move the case statement to an instance method in Amber::Exceptions and do a catch all there. That will clean this handler and there is no need to create and exception handler. We already have a handler that renders 404 and 500 errors. That can be called after the catch all